src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save movie: " + movie + " into DB: ", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        Movie movie = null;
+        try (SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```

Comment:
u should not close a sessionFactory in ur DAO methods, so lets move it out of try block
https://stackoverflow.com/questions/30096416/where-to-open-and-where-to-close-sessionfactory-in-my-application#:~:text=But%20please%20do%20not%20close%20the%20session%20factory%20at%20the%20end%20of%20a%20DAO%20method.